### DISCUSSION TITLE
Using plugin for wazuh-dashboard

### DISCUSSION THREAD

**Discussion by ghost** [wazuh/wazuh-kibana-app#6204] [General]
Please guide me how to build wazuh-dashboard plugin in wazuh dashboard. 

I tried building wazuh-dashboard 4.7.
1. git clone
2. yarn osd bootstrap
3. yarn opensearch snapshot (another terminal)
4. yarn start

Using these commands wazuh-dashboard shows on port 5603 but I am unable to build wazuh plugin

1. I cloned wazuh-dashboard-plugin 4.7 inside plugins folder of wazuh-dashboard
2. yarn build
(Here I am getting multiple conflicting dependencies error, related to webpack package)

Would love to provide any other info, if required.. kindly guide.

Also, does wazuh-dashboard-plugin integration only works with kibana(as the base)? Because I am using wazuh-dashboard. I also tried with opensearch dashboard but it also failed (same method). 
In the latest released it is mentioned **Wazuh plugins for Wazuh dashboard v4.7.0-2.8.0**

**Comment by Tostti** [comment]
Hello @adityaSTL, apologies for the late response.

This repo contains the Wazuh plugin that already come installed in the Wazuh Dashboard packages, it is the `Wazuh` section.

![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/42900763/21f212c9-b7cd-47c5-bb61-7ea38e285921)

If you need to manually build the plugin for some reason, there are 2 ways of doing it: 

-  Inside `wazuh-dashboard-plugin/plugins/main` run `yarn build:runner` (it uses Docker). This will bring up a container that builds the plugin. If the process is successful, you will find the package in `wazuh-dashboard-plugin/plugins/main/build`

- The second way is to clone the  https://github.com/opensearch-project/OpenSearch-Dashboards repo (the tag to use is the  `pluginPlatform.version` in https://github.com/wazuh/wazuh-dashboard-plugins/blob/v4.7.3-2.8.0/plugins/main/package.json. It may change depending on the Wazuh version that you are using). After that, copy the `wazuh-dashboard-plugin/plugins/main` folder inside of  `OpenSearch-Dashboards/plugins`, then `yarn osd bootstrap` and `yarn build`

I hope you find this information helpful!
